python/speech-to-text/main.py

```diff
@@ -0,0 +1,83 @@
+import os
+import json
+import wave
+import base64
+from googleMain import google_speech_recognition
+from amazonMain import amazon_recognize
+from azureSpeech import azure_recognize
+
+def write_wav(audioBase64, fileName):
+    base_path = os.path.dirname(os.path.abspath(__file__))
+    filePath = os.path.join(base_path, "tempAudio",fileName)
+
+    try :
+        wav_file = open(filePath, "wb")
```

Comment:
when you open a file like this, you need to make sure to close. best practice when opening a file is actually to with a context like:

```python
with open(filePath, "wb") as wav_file:
  wav_file.write(...)
```

For more info, see https://docs.python.org/3/tutorial/inputoutput.html#tut-files